blocks/library/embed/index.js

```diff
@@ -458,5 +470,11 @@ registerBlockType(
 	getEmbedBlockSettings( {
 		title: 'YouTube',
 		icon: 'video-alt3',
+		name: 'core-embed/youtube',
```

Comment:
With some duplication here on block name, especially specific to the cases where we're providing patterns, do you have any thoughts on my suggestion at https://github.com/WordPress/gutenberg/pull/2175#pullrequestreview-54103820 for consolidating embed registration.

Maybe:

```js
[
	[ 'core/embed', 'Embed', 'video-alt3' ],
	[ 'core-embed/twitter', 'Twitter', 'twitter' ],
	[
		'core-embed/youtube',
		'YouTube',
		'video-alt3',
		[
			/^\s*(https?:\/\/(?:m\.|www\.)?youtube\.com\/watch\S+)\s*/i,
			/^\s*(https?:\/\/(?:m\.|www\.)?youtube\.com\/playlist\S+)\s*/i,
			/^\s*(https?:\/\/youtu\.be\/\S+)\s*/i,
		],
	],
	// ...
].forEach( ( [ blockName, title, icon, patterns ] ) => {
	registerBlockType(
		blockName,
		getEmbedBlockSettings( {
			title,
			icon,
			blockName,
			patterns,
		} )
	);
} );
```

(Or instead of array structure, an object `{ blockName, title, icon, patterns }`)

My main concern being the implied dependency between `name` and `matchers`. Since `name` is only being passed to `getEmbedBlockSettings` if `matchers` passed as well, this might mislead future maintainers to think they have access to name in future revisions to `getEmbedBlockSettings`. Or, otherwise, we pass `name` in all cases, but then we have excessive duplication.